packaging/installer/kickstart-ng.sh

```diff
@@ -0,0 +1,843 @@
+  DO_NOT_TRACK               If set to a value other than 0, behave as if \`--disable-telemetry\` was specified.
+
+HEREDOC
+}
+
+# ======================================================================
+# Utility functions
+
+setup_terminal() {
+  TPUT_RESET=""
+  TPUT_WHITE=""
+  TPUT_BGRED=""
+  TPUT_BGGREEN=""
+  TPUT_BOLD=""
+  TPUT_DIM=""
+
+  # Is stderr on the terminal? If not, then fail
+  test -t 2 || return 1
+
+  if command -v tput > /dev/null 2>&1; then
+    if [ $(($(tput colors 2> /dev/null))) -ge 8 ]; then
+      # Enable colors
+      TPUT_RESET="$(tput sgr 0)"
+      TPUT_WHITE="$(tput setaf 7)"
+      TPUT_BGRED="$(tput setab 1)"
+      TPUT_BGGREEN="$(tput setab 2)"
+      TPUT_BOLD="$(tput bold)"
+      TPUT_DIM="$(tput dim)"
+    fi
+  fi
+
+  echo "${TPUT_RESET}"
+
+  return 0
+}
+
+fatal() {
+  printf >&2 "%s\n\n" "${TPUT_BGRED}${TPUT_WHITE}${TPUT_BOLD} ABORTED ${TPUT_RESET} ${*}"
+  exit 1
+}
+
+run_ok() {
+  printf >&2 "%s\n\n" "${TPUT_BGGREEN}${TPUT_WHITE}${TPUT_BOLD} OK ${TPUT_RESET}"
+}
+
+run_failed() {
+  printf >&2 "%s\n\n" "${TPUT_BGRED}${TPUT_WHITE}${TPUT_BOLD} FAILED ${TPUT_RESET}"
+}
+
+ESCAPED_PRINT_METHOD=
+# shellcheck disable=SC3050
+if printf "%q " test > /dev/null 2>&1; then
+  ESCAPED_PRINT_METHOD="printfq"
+fi
+
+escaped_print() {
+  if [ "${ESCAPED_PRINT_METHOD}" = "printfq" ]; then
+    # shellcheck disable=SC3050
+    printf "%q " "${@}"
+  else
+    printf "%s" "${*}"
+  fi
+  return 0
+}
+
+progress() {
+  echo >&2 " --- ${TPUT_BOLD}${*}${TPUT_RESET} --- "
+}
+
+run_logfile="/dev/null"
+run() {
+  user="${USER--}"
+  dir="${PWD}"
+
+  if [ "$(id -u)" = "0" ]; then
+    info="[root ${dir}]# "
+    info_console="[${TPUT_DIM}${dir}${TPUT_RESET}]# "
+  else
+    info="[${user} ${dir}]$ "
+    info_console="[${TPUT_DIM}${dir}${TPUT_RESET}]$ "
+  fi
+
+  {
+    printf "%s" "${info}"
+    escaped_print "${@}"
+    printf " ... "
+  } >> "${run_logfile}"
+
+  printf >&2 "%s" "${info_console}${TPUT_BOLD}"
+  escaped_print >&2 "${@}"
+  printf >&2 "%s\n" "${TPUT_RESET}"
+
+  "${@}"
+
+  ret=$?
+  if [ ${ret} -ne 0 ]; then
+    run_failed
+    printf "%s\n" "FAILED with exit code ${ret}" >> "${run_logfile}"
+  else
+    run_ok
+    printf "OK\n" >> "${run_logfile}"
+  fi
+
+  return ${ret}
+}
+
+fatal() {
+  printf >&2 "%s\n\n" "${TPUT_BGRED}${TPUT_WHITE}${TPUT_BOLD} ABORTED ${TPUT_RESET} ${*}"
+  exit 1
+}
+
+warning() {
+  printf >&2 "%s\n\n" "${TPUT_BGRED}${TPUT_WHITE}${TPUT_BOLD} WARNING ${TPUT_RESET} ${*}"
+}
+
+_cannot_use_tmpdir() {
+  testfile="$(TMPDIR="${1}" mktemp -q -t netdata-test.XXXXXXXXXX)"
+  ret=0
+
+  if [ -z "${testfile}" ] ; then
+    return "${ret}"
+  fi
+
+  if printf '#!/bin/sh\necho SUCCESS\n' > "${testfile}" ; then
+    if chmod +x "${testfile}" ; then
+      if [ "$("${testfile}")" = "SUCCESS" ] ; then
+        ret=1
+      fi
+    fi
+  fi
+
+  rm -f "${testfile}"
+  return "${ret}"
+}
+
+create_tmp_directory() {
+  if [ -z "${TMPDIR}" ] || _cannot_use_tmpdir "${TMPDIR}" ; then
```

Comment:
Formatting - sometimes you add a space before `;`, sometimes don't. From what I remember we don't add it.

(that is not the only case with a space befoe `;`)

Reply:
Yeah, this is a side-effect of pulling some existing code from multiple sources, and my ‘instinctive’ style not matching up with our official styling.